src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -35,6 +35,16 @@ namespace NUnit.Engine.Services
     /// </summary>
     public class DefaultTestRunnerFactory : InProcessTestRunnerFactory, ITestRunnerFactory
     {
+        private IProjectService _projectService;
+
+        public override void StartService()
+        {
+            _projectService = ServiceContext.GetService<IProjectService>();
+            Status = _projectService != null && ((IService)_projectService).Status == ServiceStatus.Started
```

Comment:
Took a look. What's preventing this change is that a few services don't have an interface. I think I'll add them. 

DomainManager, which is used internally, has never used an interface. Adding one does no harm and makes it consistent with other services. It will be fakeable as well, which could be useful.

SettingsService uses the ISettings interface, but doesn't implement it directly. It's implemented by SettingsGroup, from which it is indirectly derived. I'll look at how much effort I want to put into simplifying this. FWIW, the structure of these classes goes back to a time when NUnit supported settings stored in multiple ways, including in the registry.